lib/db/mongodb/errors.go

```diff
@@ -9,15 +9,16 @@ const (
 )
 
 func HasErrorCode(err error, code int) bool {
-	we, ok := err.(mongo.WriteException)
-	if !ok {
-		return false
-	}
-
-	for _, err := range we.WriteErrors {
-		if err.Code == code {
-			return true
+	switch v := err.(type) {
+	case mongo.WriteException:
+	case mongo.BulkWriteException:
+		for _, err := range v.WriteErrors {
```

Comment:
Are you absolutely sure these will be the same on both? I recall there possibly being some additional wrapping for one of them (i.e. you might need to go down a level further or something)

Reply:
Looks 💯 to add support for them tho

Reply:
Looks like `BulkWriteException` is being wrapped the same way as for `WriteException`: https://github.com/mongodb/mongo-go-driver/blob/cc85d3210f49d9daef608a2c61cdd3f490514e9a/mongo/integration/unified/error_test.go#L137

Now my concern is with the style: matching same block of logic within two cases one by one or duplicate the logic for each case (as in referred mongo source code).

Reply:
Nice, looks that way!